src/gval/utils/loading_datasets.py

```diff
@@ -238,205 +236,81 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
         return xr_object
 
 
-def _get_raster_band_nodata(band_metadata, nodata_fill) -> Number:
-    """
-    Extracts nodata information from STAC APIs that implement Raster Extension
+def stac_to_df(
+    stac_items: ItemCollection,
+    assets: list = None,
+    attribute_allow_list: list = None,
+    attribute_block_list: list = None,
+) -> pd.DataFrame:
+    """Convert STAC Items in to a DataFrame
 
     Parameters
     ----------
-    band_metadata : list
-        Metadata from raster:bands extension
-    nodata_fill : Number
-        Fill in value for missing data
+    stac_items: ItemCollection
+        STAC Item Collection returned from pystac client
+    assets : list, default = None
+        Assets to keep, (keep all if None)
+    attribute_allow_list: list, default = None
+        List of columns to allow in the result DataFrame
+    attribute_block_list: list, default = None
+        List of columns to remove in the result DataFrame
 
     Returns
     -------
-    Number
-        Number representing nodata
+    pd.DataFrame
+        A DataFrame with rows for each unique item/asset combination
 
     Raises
     ------
     ValueError
+        No entries in DataFrame due to nonexistent asset
 
     """
 
-    if band_metadata:
-        prop_string = str(band_metadata.coords["raster:bands"].values)
-        idx1, idx2 = prop_string.find("{"), prop_string.rfind("}")
-
-        return ast.literal_eval(prop_string[idx1 : idx2 + 1]).get("nodata")
-    else:
-        if nodata_fill is None:
-            raise ValueError(
-                "Must have nodata fill value if nodata is not present in metadata"
-            )
-
-        return nodata_fill
-
-
-def _set_nodata(
-    stack: xr.DataArray, band_metadata: list = None, nodata_fill: Number = None
-) -> Number:
-    """
-    Sets nodata information from STAC APIs that implement Raster Extension
-
-    Parameters
-    ----------
-    stack : xr.DataArray
-        Data to set nodata attribute
-    band_metadata : list
-        Metadata from raster:bands extension
-    nodata_fill : Number
-        Fill in value for missing data
-
-    """
-
-    if stack.rio.nodata is not None:
-        stack.rio.write_nodata(stack.rio.nodata, inplace=True)
-    elif stack.rio.encoded_nodata is not None:
-        stack.rio.write_nodata(stack.rio.encoded_nodata, inplace=True)
-    else:
-        stack.rio.write_nodata(
-            _get_raster_band_nodata(band_metadata, nodata_fill), inplace=True
-        )
-
+    item_dfs, compare_idx = [], 1
 
-def _set_crs(stack: xr.DataArray, band_metadata: list = None) -> Number:
-    """
+    # Iterate through each STAC Item and make a unique row for each asset
+    for item in stac_items:
+        item_dict = item.to_dict()
+        item_df = pd.json_normalize(item_dict)
+        mask = item_df.columns.str.contains("assets.*")
```

Comment:
What does this `assets.*` line do?

Reply:
As discussed they remove asset columns so there can be a row for each asset.